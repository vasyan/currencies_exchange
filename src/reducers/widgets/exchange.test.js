import reducer from './exchange'

describe('Exchange reducer', () => {
  it('should return initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  it('should change currency for exchange', () => {
    expect(
      reducer(undefined, { type: 'exchange/CHANGE_CURRENCY_FROM', payload: 1 })
    ).toMatchSnapshot()
    expect(
      reducer(undefined, { type: 'exchange/CHANGE_CURRENCY_FROM', payload: -1 })
    ).toMatchSnapshot()
  })

  it('should change currency for exchange to', () => {
    expect(
      reducer(undefined, { type: 'exchange/CHANGE_CURRENCY_TO', payload: 1 })
    ).toMatchSnapshot()
    expect(
      reducer(undefined, { type: 'exchange/CHANGE_CURRENCY_TO', payload: -1 })
    ).toMatchSnapshot()
  })
})
