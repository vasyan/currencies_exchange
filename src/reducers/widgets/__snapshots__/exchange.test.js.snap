// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange reducer should change currency for exchange 1`] = `
Object {
  "currencyFrom": "EUR",
  "currencyTo": "EUR",
  "input": null,
  "output": null,
}
`;

exports[`Exchange reducer should change currency for exchange 2`] = `
Object {
  "currencyFrom": "GBP",
  "currencyTo": "EUR",
  "input": null,
  "output": null,
}
`;

exports[`Exchange reducer should change currency for exchange to 1`] = `
Object {
  "currencyFrom": "USD",
  "currencyTo": "GBP",
  "input": null,
  "output": null,
}
`;

exports[`Exchange reducer should change currency for exchange to 2`] = `
Object {
  "currencyFrom": "USD",
  "currencyTo": "USD",
  "input": null,
  "output": null,
}
`;

exports[`Exchange reducer should return initial state 1`] = `
Object {
  "currencyFrom": "USD",
  "currencyTo": "EUR",
  "input": null,
  "output": null,
}
`;
